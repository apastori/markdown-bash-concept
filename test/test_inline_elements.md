
# Inline Elements Test

This file tests the inline elements.

## Emphasis

This is `inline code`.
This is *italic* and this is _italic_.
This is **bold** and this is __bold__.
This is ***bold and italic*** and this is ___bold and italic___.

This is a mix: *italic*, **bold**, `code`.

## Links and Images

This is a [link to Google](https://www.google.com).
This is an image: ![alt text](https://via.placeholder.com/150 "Title").

## Nested Elements

This is a **bold text with *italic* inside**.
This is a [link with **bold** and *italic* text](https://example.com).
This is an image with *italic* alt text: ![*alt text*](https://via.placeholder.com/150 "Title").

## Edge Cases

This is an unclosed *italic.
This is an unclosed `code.
This is an unclosed [link](https://example.com.
This is an unclosed ![image](https://via.placeholder.com/150.
