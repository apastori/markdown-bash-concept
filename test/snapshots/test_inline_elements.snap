# Tokenizer Snapshot
Tokenizing file: test/test_inline_elements.md
EMPTY_LINE||1| HEADING|1:Inline Elements Test|2| EMPTY_LINE||3| PARAGRAPH|This file tests the inline elements.|4| EMPTY_LINE||5| HEADING|2:Emphasis|6| EMPTY_LINE||7| PARAGRAPH|This is `inline code`.|8| CODE_INLINE|inline code|8|paragraph PARAGRAPH|This is *italic* and this is _italic_.|9| ITALIC|italic|9|paragraph ITALIC|italic|9|paragraph PARAGRAPH|This is **bold** and this is __bold__.|10| BOLD|bold|10|paragraph BOLD|bold|10|paragraph PARAGRAPH|This is ***bold and italic*** and this is ___bold and italic___.|11| BOLD_ITALIC|bold and italic|11|paragraph BOLD_ITALIC|bold and italic|11|paragraph EMPTY_LINE||12| PARAGRAPH|This is a mix: *italic*, **bold**, `code`.|13| ITALIC|italic|13|paragraph BOLD|bold|13|paragraph CODE_INLINE|code|13|paragraph EMPTY_LINE||14| HEADING|2:Links and Images|15| EMPTY_LINE||16| PARAGRAPH|This is a [link to Google](https://www.google.com).|17| LINK|link to Google:https://www.google.com|17|paragraph PARAGRAPH|This is an image: ![alt text](https://via.placeholder.com/150 "Title").|18| IMAGE|alt text:https://via.placeholder.com/150 "Title":|18|paragraph EMPTY_LINE||19| HEADING|2:Nested Elements|20| EMPTY_LINE||21| PARAGRAPH|This is a **bold text with *italic* inside**.|22| BOLD|bold text with *italic* inside|22|paragraph ITALIC|italic|22|paragraph_nested PARAGRAPH|This is a [link with **bold** and *italic* text](https://example.com).|23| LINK|link with **bold** and *italic* text:https://example.com|23|paragraph BOLD|bold|23|link_text ITALIC|italic|23|link_text PARAGRAPH|This is an image with *italic* alt text: ![*alt text*](https://via.placeholder.com/150 "Title").|24| ITALIC|italic|24|paragraph IMAGE|*alt text*:https://via.placeholder.com/150 "Title":|24|paragraph ITALIC|alt text|24|image_alt EMPTY_LINE||25| HEADING|2:Edge Cases|26| EMPTY_LINE||27| PARAGRAPH|This is an unclosed *italic.|28| TEXT|*|28|paragraph PARAGRAPH|This is an unclosed `code.|29| TEXT|`|29|paragraph PARAGRAPH|This is an unclosed [link](https://example.com.|30| PARAGRAPH|This is an unclosed ![image](https://via.placeholder.com/150.|31| TEXT|!|31|paragraph
